core/src/main/java/cucumber/runtime/formatter/JSONFormatter.java

```diff
@@ -48,7 +49,8 @@
     private final Gson gson = new GsonBuilder().setPrettyPrinting().create();
     private final NiceAppendable out;
     private final TestSourcesModel testSources = new TestSourcesModel();
-
+    private final TimeUtils timeUtils;
```

Comment:
I don't think this is needed. The event bus is constructed with a time provider that allows tests to receive deterministic time (kinda like `Clock` from Java 8). If you implement `TimeServiceStub.timeStampMillis` in the same way as `TimeServiceStub.timeStamp`.

Reply:
I imlemented timeStampMillis in TimeServiceStub as well but there are a number of JUnits which don't pass an eventBus to the Runner.build() because they do not expect the 'duration' field in the response. But now they would have the timeStampMillis field since its at the scenario level.

For eg : **RuntimeTest.runs_feature_with_json_formatter()**

For these Junits I had to add the TimeUtils so i could mock it. Should I update the JUnits instead to use an eventBus with TimeServiceStub?

Reply:
Please do! Unless it's too much work. Then we can look into making it easier to use the time service.

Reply:
No I was actually hoping I could do that :) Just wanted to make sure there's no issue if i update them instead.

Let me check and I ll udpate.

Reply:
Done